instrumentation/rabbitmq-2.7/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/rabbitmq/RabbitMqTest.java

```diff
@@ -0,0 +1,955 @@
+                  });
+        });
+    traceAssertions.add(
+        trace -> {
+          trace
+              .hasSize(1)
+              .hasSpansSatisfyingExactly(
+                  span -> {
+                    rabbitSpan(trace, span, 0, null, null, null, "basic.consume");
+                  });
+        });
+
+    for (int i = 1; i <= messageCount; i++) {
+      traceAssertions.add(
+          trace -> {
+            trace
+                .hasSize(2)
+                .hasSpansSatisfyingExactly(
+                    span -> {
+                      rabbitSpan(trace, span, 0, exchangeName, null, "publish", exchangeName);
+                    },
+                    span -> {
+                      rabbitSpan(
+                          trace,
+                          span,
+                          1,
+                          exchangeName,
+                          null,
+                          "process",
+                          resource,
+                          trace.getSpan(0),
+                          null,
+                          null,
+                          null,
+                          setTimestamp,
+                          false);
+                    });
+          });
+    }
+
+    testing.waitAndAssertTraces(traceAssertions);
+
+    assertEquals(messageCount, deliveries.size());
+    for (int i = 1; i <= messageCount; i++) {
+      assertEquals("msg " + i, deliveries.get(i - 1));
+    }
+  }
+
+  @Test
+  void testRabbitConsumeError() throws IOException {
+    String exchangeName = "some-error-exchange";
+    FileNotFoundException error = new FileNotFoundException("Message Error");
+    channel.exchangeDeclare(exchangeName, "direct", false);
+    String queueName = channel.queueDeclare().getQueue();
+    channel.queueBind(queueName, exchangeName, "");
+
+    Consumer callback =
+        new DefaultConsumer(channel) {
+          @Override
+          public void handleDelivery(
+              String consumerTag, Envelope envelope, AMQP.BasicProperties properties, byte[] body)
+              throws IOException {
+            throw error;
+            // Unfortunately this doesn't seem to be observable in the test outside of the span
+            // generated.
+          }
+        };
+
+    channel.basicConsume(queueName, callback);
+
+    channel.basicPublish(exchangeName, "", null, "msg".getBytes(Charset.defaultCharset()));
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace
+                .hasSize(1)
+                .hasSpansSatisfyingExactly(
+                    span -> {
+                      rabbitSpan(trace, span, 0, null, null, null, "exchange.declare");
+                    }),
+        trace ->
+            trace
+                .hasSize(1)
+                .hasSpansSatisfyingExactly(
+                    span -> {
+                      rabbitSpan(trace, span, 0, null, null, null, "queue.declare");
```

Comment:
would another overload for this `0, null, null, null` case help a little with readability here?